src/devices/tuya.ts

```diff
@@ -4167,65 +4167,97 @@ const definitions: Definition[] = [
-        toZigbee: [tuya.tz.datapoints],
+        description: 'Avatto wall thermostat',
         onEvent: tuya.onEvent({timeStart: '1970'}),
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
         configure: tuya.configureMagicPacket,
         exposes: [
+            e.binary('factory_reset', ea.STATE_SET, 'ON', 'OFF')
+                .withDescription('Full factory reset, use with caution!'),
             e.child_lock(),
             e.climate()
                 .withSystemMode(['off', 'heat'], ea.STATE_SET)
-                .withPreset(['auto', 'manual', 'temporary program'])
+                .withPreset(['auto', 'manual', 'temporary_manual'])
                 .withSetpoint('current_heating_setpoint', 5, 35, 0.5, ea.STATE_SET)
                 .withRunningState(['idle', 'heat'], ea.STATE)
                 .withLocalTemperature(ea.STATE)
                 .withLocalTemperatureCalibration(-9.9, 9.9, 0.1, ea.STATE_SET),
             e.binary('frost_protection', ea.STATE_SET, 'ON', 'OFF')
                 .withDescription('Antifreeze function'),
-            e.numeric('upper_temp', ea.STATE_SET).withUnit('°C').withValueMax(95)
-                .withValueMin(15).withValueStep(1).withPreset('default', 60, 'Default value')
+            e.max_temperature_limit()
+                .withUnit('°C')
+                .withValueMin(15)
+                .withValueMax(90)
+                .withValueStep(0.5)
+                .withPreset('default', 60, 'Default value')
                 .withDescription('Maximum upper temperature'),
-            e.numeric('deadzone_temperature', ea.STATE_SET).withUnit('°C').withValueMax(10)
-                .withValueMin(0.5).withValueStep(0.5).withPreset('default', 1, 'Default value')
-                .withDescription('The delta between local_temperature and current_heating_setpoint to trigger Heat'),
-            // Not working yet:
-            // e.enum('schedule_mode', ea.STATE_SET, ['disabled','weekday/sat+sun','weekday+sat/sun','7day'])
-            //     .withDescription('Schedule mode')
+            e.numeric('deadzone_temperature', ea.STATE_SET)
+                .withUnit('°C')
+                .withValueMax(10)
+                .withValueMin(0.5)
+                .withValueStep(0.5)
+                .withPreset('default', 1, 'Default value')
+                .withDescription('The delta between local_temperature (5<t<35)and current_heating_setpoint to trigger Heat'),
+            e.enum('backlight_mode', ea.STATE_SET, ['off', 'low', 'medium', 'high'])
+                .withDescription('Intensity of the backlight'),
+            e.enum('working_day', ea.STATE_SET, ['disabled', '6-1', '5-2', '7'])
+                .withDescription('Workday setting'),
+            e.text('schedule_weekday', ea.STATE_SET).withDescription('Workdays (6 times `hh:mm/cc.c°C`)'),
+            e.text('schedule_holiday', ea.STATE_SET).withDescription('Holidays (2 times `hh:mm/cc.c°C)`'),
```

Comment:
I already merged this to be in time for the 1 December release, but I think it will be good to add an example to the description, e.g.

```
e.text('schedule_holiday', ea.STATE_SET).withDescription('Holidays (2 times `hh:mm/cc.c°C`, e.g. `02:04/18.0°C,18:20/19.0°C`)'),
```

Reply:
I will ..
Tested it with ZIgbee2MQTT-edge.
It seems you missed somehow some changes in /lib/tuya.ts:

![image](https://github.com/Koenkk/zigbee-herdsman-converters/assets/51886888/dd114b4f-320a-4cbf-84de-72e28f833532)

theré I add a datapoint key 'schedule_holiday' and in the merge (and the Edge-version that's missing

Reply:
fixed!